umbraco/Umbraco-CMS
# Netcore: Individual nuget  packages for each assembly

### Description
This PR switches the build procedure to use `dotnet pack` which creates a separate nuget package for each project in the solution. 

Dependencies are handles through project references. UmbracoCms.nuspec has been changed to have a dependency to `Umbraco.Cms.Web.Website` and `Umbraco.Cms.Web.BackOffice`, all other projects will then be included as implicit references.

### Test procedure
To test the pr: 
1. Enter the root folder in a powershell prompt
2. Run the build with `.\build\build.ps1`
   1. Once the build is done you can see the nuget packages in the `build.out` folder
2. Open a normal terminal and enter the folder where you want to create the test project. 
3. To be on the safe side I'd recommend clearing you nuget cache with `dotnet nuget locals all --clear`
4. Add the Examine-Beta pipeline with `dotnet nuget add source https://shazwazza.pkgs.visualstudio.com/Examine/_packaging/Examine-Beta/nuget/v3/index.json "ExamineAzurePipelines"`
5. Add the `build.out` folder as a local nuget feed with: `dotnet nuget add source <Path\to\build.out> -n "LocalFeed"`
6. Install the template: `dotnet new -i Umbraco.Templates::9.0.0-alpha004`
7. Create a project from the template: `dotnet new umbraco -n <projectName>`
   1. To test it with SqlCE do `dotnet new umbraco -U true -n <projectName>`

Ensure that the project builds and run and that the correct dependencies are installed, and that the metadata has been updated.
